core/types/state_account.go

```diff
@@ -32,3 +34,65 @@ type StateAccount struct {
 	Root     common.Hash // merkle root of the storage trie
 	CodeHash []byte
 }
+
+// SlimAccount is a modified version of an Account, where the root is replaced
+// with a byte slice. This format can be used to represent full-consensus format
+// or slim format which replaces the empty root and code hash as nil byte slice.
+type SlimAccount struct {
+	Nonce    uint64
+	Balance  *big.Int
+	Root     []byte // Nil if root equals to types.EmptyRootHash
+	CodeHash []byte // Nil if hash equals to types.EmptyCodeHash
+}
+
+// SlimAccountRLP encodes the state account in 'slim RLP' format.
+func SlimAccountRLP(account StateAccount) []byte {
+	slim := SlimAccount{
+		Nonce:   account.Nonce,
+		Balance: account.Balance,
+	}
+	if account.Root != EmptyRootHash {
+		slim.Root = account.Root[:]
+	}
+	if !bytes.Equal(account.CodeHash, EmptyCodeHash[:]) {
+		slim.CodeHash = account.CodeHash
+	}
+	data, err := rlp.EncodeToBytes(slim)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
This `panic` here can probably be omitted, -- we can ignore the `err`, right?

Reply:
We can omit the error and commit "rlp encode can never be failed".

Reply:
I'm also fine with either (hence I approved it), but since rlp encoding cannot fail, I think we should omit the panic

Reply:
I found another case that RLP encode can fail.

```go

var acct types.StateAccount
rlp.EncodeBytes(acct)
```

In this case, pointer of `types.StateAccount` is required as the parameter. I just want to say that rlp may or may not will fail, but panic is better than ignore error silently.